packages/rest/src/rest-component.ts

```diff
@@ -0,0 +1,54 @@
+// Copyright IBM Corp. 2017. All Rights Reserved.
+// Node module: @loopback/rest
+// This file is licensed under the MIT License.
+// License text available at https://opensource.org/licenses/MIT
+
+import {
+  ApplicationConfig,
+  Component,
+  CoreBindings,
+  ProviderMap,
+  Server,
+  Application,
+} from '@loopback/core';
+import {
+  BoundValue,
+  Context,
+  Injection,
+  ValueOrPromise,
+  inject,
+  Constructor,
+} from '@loopback/context';
+import {RestBindings} from './keys';
+import {LogErrorProvider} from './providers';
+import {RestServer, RestServerConfig} from './rest-server';
+
+export class RestComponent implements Component {
+  providers: ProviderMap = {
+    [RestBindings.SequenceActions.LOG_ERROR]: LogErrorProvider,
+  };
+  servers: Constructor<Server>[] = [];
+  constructor(
+    @inject(CoreBindings.APPLICATION_INSTANCE) app: Application,
+    @inject(CoreBindings.APPLICATION_CONFIG, undefined, getRestConfig)
```

Comment:
Is this a workaround for hash-keyed binding not resolving correctly for you? I think we should sort this out before landing this pull request.

Reply:
I managed to track down this problem. It's a bug in the implementation of hash-keyed properties. The support for hash keys was implemented in `ctx.get`/`ctx.getSync` only (see https://github.com/strongloop/loopback-next/pull/377). However, `@inject` uses different code path and invokes `binding.getValue` directly - see https://github.com/strongloop/loopback-next/blob/1bdccc874748290c5757a140e3deace8ed86d87b/packages/context/src/resolver.ts#L77-L79

As a result, the code resolving nested property keys is not invoked.

Note that in this pull request, you have code invoking `binding.getValue` too.

I think we need to fix the resolution of nested properties to happen at a binding level, to ensure that `binding.getValue` returns the requested nested property too.

The fix should be done in a new/standalone pull request, this one is already too large and takes too long to finish. I'll start looking into this myself right now.

Reply:
Here is the pull request fixing this problem for reference: https://github.com/strongloop/loopback-next/pull/587